blocks/library/image/style.scss

```diff
@@ -2,7 +2,7 @@
 	&[data-align="left"],
 	&[data-align="right"] {
 		// Without z-index, won't be clickable as "above" adjacent content
-		z-index: 10;
+		z-index: z-index( '.editor-visual-editor__block' );
```

Comment:
In general the names in the `z-index` map should correspond very closely to the selectors that apply each rule.  This one is not really `.editor-visual-editor__block` - it should be something more like this:

```css
.editor-visual-editor__block[data-type="core/image"][data-align]
```

In this case it might be a good idea to clarify the name a bit:

```css
.editor-visual-editor__block {core/image aligned left or right}
```